packages/lib/response/tests/__mocks__/data.mock.ts

```diff
@@ -0,0 +1,96 @@
+import { Prisma } from "@prisma/client";
+import { responseNoteSelect } from "../../../responseNote/service";
+import { responseSelection } from "../../service";
+import { randomCuid2 } from "../../../utils/common";
+import { randBoolean, randBrowser, randFullName, randText, randUrl, randUuid } from "../constants";
+
+type ResponseMock = Prisma.ResponseGetPayload<{
+  include: typeof responseSelection;
+}>;
+type ResponseNoteMock = Prisma.ResponseNoteGetPayload<{
+  include: typeof responseNoteSelect;
+}>;
+type ResponsePersonMock = Prisma.PersonGetPayload<{
+  select: typeof responseSelection.person.select;
+}>;
+
+export const mockEnvironmentId = randomCuid2();
```

Comment:
@pandeymangg I think we shouldn't use random values in the tests since this would make the test non-deterministic:

---

Using random values in unit tests, like your `randBrowser()` and `randBoolean()` functions, can lead to non-deterministic tests, which might not be ideal in many testing scenarios. Here's a breakdown of the considerations:

### Non-Deterministic Tests: Drawbacks

1. **Reproducibility**: If a test fails due to a specific random value, it can be challenging to reproduce and debug the issue since the same value may not be generated again.
2. **Flakiness**: Tests might pass sometimes and fail at others, depending on the random values generated. This can lead to uncertainty about the stability and reliability of the code.
3. **Coverage Inconsistency**: Random values might not consistently cover all the edge cases or paths in the code, leading to inconsistent test coverage.

### Deterministic Tests: Advantages

1. **Reproducibility**: Tests produce the same results every time, making it easier to identify and fix issues.
2. **Reliability**: Consistent outcomes increase confidence in the test results and the code's stability.
3. **Complete Coverage**: You can design tests to cover all relevant cases and edge cases, ensuring comprehensive testing.

### When to Use Randomness

Using randomness in tests can be beneficial in certain scenarios:

1. **Stress Testing**: To test how the system behaves under varied and unpredictable inputs.
2. **Fuzz Testing**: To identify potential crashes or unexpected behavior in the system.
3. **Large Data Sets**: When dealing with huge datasets, random sampling might be necessary for practicality.

### Best Practices

- **Controlled Randomness**: If you use random values, consider using a seeded random number generator. This way, the "random" values are the same across different test runs, maintaining some level of determinism.
- **Specific Test Cases for Edge Conditions**: Write separate deterministic tests for edge cases and specific conditions you want to ensure are always tested.
- **Monitoring Flaky Tests**: If you use randomness, monitor for flakiness in your test suite, and investigate tests that show inconsistent results.

### Conclusion

In your context, if the random values are not creating flakiness or missing coverage, they might be acceptable. However, generally, it's better to have deterministic tests, especially for unit testing, where you're testing specific functions or components under known conditions. For broader system testing or in specific scenarios like stress or fuzz testing, randomness can be more appropriate.

Reply:
okay, should we then hard code these values?